lib/collection/src/shards/shard_holder/resharding.rs

```diff
@@ -344,51 +344,55 @@ impl ShardHolder {
         }
 
         // Remove new shard if resharding up
-        if direction == ReshardingDirection::Up {
+        if (is_in_progress || force) && direction == ReshardingDirection::Up {
```

Comment:
I don't think we can use `force` here. It will basically allow you to remove any shard, if you abort resharding with proper `reshard_key` and a `force` flag.

Reply:
The `force` is intended to allow aborting if we are at `ReadHashRingCommitted`/`WriteHashRingCommitted` stage.

Reply:
> It will basically allow you to remove any shard, if you abort resharding with proper reshard_key and a force flag.

Isn't that kind of the point. The reshard key should only match a new shard that is newly created with resharding, and it can therefore not be used to remove an arbitrary shard ID.

> The force is intended to allow aborting if we are at ReadHashRingCommitted/WriteHashRingCommitted stage.

I agree that that was the original intention. But I was looking at it from a 'will this help us resolve corrupt state' perspective. In that case I think allowing this as well will be useful.

What do you think?

Reply:
> The reshard key should only match a new shard that is newly created with resharding, and it can therefore not be used to remove an arbitrary shard ID.

As far as I can tell, there's nothing in `abort_resharding` that prevents calling it with garbage `ReshardKey` (e.g., when resharding is not in progress or with incorrect `shard_id`). And it such case, we will just purge the whole shard no questions asked.

Reply:
You're right. With the current state in the PR it is possible to bypass the shard key check with the force flag.

I agree that isn't desirable so I'll adjust.